.github/workflows/ci-casper-rust-contract.yml

```diff
@@ -9,68 +9,134 @@ on:
-      # Needed for gcc install
       - run: sudo apt update && sudo apt install -y build-essential wabt
-      - uses: Swatinem/rust-cache@v1
+      - uses: Swatinem/rust-cache@v2
       - run: make prepare
       - run: make check-lint
       - run: make test
+      - name: Install cargo-audit
+        run: cargo install cargo-audit
+      - name: Run cargo audit
+        run: cargo audit
       - uses: actions/upload-artifact@v4
         with:
           name: cep18.wasm
           path: target/wasm32-unknown-unknown/release/cep18.wasm
 
-  # ci-js-client:
-  #   needs: ci-smart-contract
-  #   strategy:
-  #     matrix:
-  #       # See supported Node.js release schedule at https://nodejs.org/en/about/releases/
-  #       node-version: [20.x, 22.x]
-  #   runs-on: ubuntu-22.04
-  #   env:
-  #     NODE_URL: http://127.0.0.1:7777/rpc
-  #     EVENT_STREAM_ADDRESS: http://127.0.0.1:9999/events/main
-
-  #   steps:
-  #     - name: Checkout
-  #       uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b #v3.0.2
-  #     - uses: actions/download-artifact@v4
-  #       with:
-  #         name: cep18.wasm
-  #         path: target/wasm32-unknown-unknown/release
-
-  #  Disable client-js testing
-  # - name: Use Node.js ${{ matrix.node-version }}
-  #   uses: actions/setup-node@5b949b50c3461bbcd5a540b150c368278160234a #v3.4.0
-  #   with:
-  #     node-version: ${{ matrix.node-version }}
-  #     registry-url: 'https://registry.npmjs.org'
-
-  # - name: Casper node launcher js
-  #   run: |
-  #     git clone --branch dist_folder https://github.com/casper-network/casper-node-launcher-js.git
-  #     cd casper-node-launcher-js
-  #     npm install
-  #     npm run build
-  #     npx casper-node-launcher-js node --daemon
-
-  # - name: Install
-  #   working-directory: client-js
-  #   run: npm install
-
-  # - name: Generate WASMS
-  #   working-directory: client-js
-  #   run: npm run generate:wasm
-
-  # - name: Integration test
-  #   working-directory: client-js
-  #   run: npm run test:e2e && npx casper-node-launcher-js stop
+  ci-js-client:
+    needs: ci-smart-contract
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        node-version: [22.x]
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - uses: actions/download-artifact@v4
+        with:
+          name: cep18.wasm
+          path: target/wasm32-unknown-unknown/release
+
+      - name: Start Casper NCTL Service
+        run: |
+          docker run -d \
+            --name casper-nctl \
+            -p 11101-11105:11101-11105 \
+            -p 14101-14105:14101-14105 \
+            -p 18101-18105:18101-18105 \
+            -p 25101-25105:25101-25105 \
+            -p 28101-28105:28101-28105 \
+            -v $PWD/assets/users:/app/casper-nctl/assets/net-1/users/ \
+            -v $PWD/assets/faucet:/app/casper-nctl/assets/net-1/faucet/ \
+            gregoshop/casper-nctl:2.0
```

Comment:
Please not nat CI/CD for e2e tests rely on a docker image https://hub.docker.com/repository/docker/gregoshop/casper-nctl/general from the RC, this might require another method to launch nctl 2.0

Reply:
https://github.com/casper-ecosystem/cep18/issues/136